Fortnox.NET/Services/AuthorizationService.cs

```diff
@@ -9,7 +14,68 @@ public class AuthorizationService
         public static async Task<AuthorizationResponse> GetAccessTokenAsync(string authorizationCode, string clientSecret)
         {
             return await FortnoxAPIClient.GetAccessTokenAsync(authorizationCode, clientSecret,
-                "https://api.fortnox.se/3/customers");
+                ApiEndpoints.Customers);
+        }
+
+        /// <summary>
+        /// Get an URL to visit in order to get the authorization code used to retrieve a access token.
+        /// </summary>
+        /// <param name="clientId">The client_id is the public identifier for the app.</param>
+        /// <param name="redirectUri">URL-encoded URI that must match the Redirect URI for the app set in the Developer Portal. If omitted, it will default to the registered Redirect URI.</param>
+        /// <param name="scopes">The request should have one or more scope values indicating access requested by the application. </param>
+        /// <param name="state">The state parameter is used by the application to store request-specific data and/or prevent CSRF attacks.</param>
+        /// <returns>A string with the URL to redirect to in order to initiate the OAuth flow.</returns>
+        public static string GetAuthorizationUrl(string clientId, string redirectUri, string scopes, string state)
+        {
+            var urlContent = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("client_id", clientId),
+                new KeyValuePair<string, string>("scope", scopes),
+                new KeyValuePair<string, string>("state", state),
+                new KeyValuePair<string, string>("access_type", "offline"),
+                new KeyValuePair<string, string>("response_type", "code"),
+            };
+
+            if (redirectUri != null)
+            {
+                urlContent.Add(new KeyValuePair<string, string>("redirect_uri", redirectUri));
+            }
+
+            var sb = new StringBuilder();
+            foreach (var kv in urlContent)
```

Comment:
Weird that there doesn't seem to exist built in support for constructing queries in .NET. There is https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.webutilities.queryhelpers.addquerystring, but seems to be some issues in regards to encoding, so doesn't really inspire confidence